### DISCUSSION TITLE
Many versions of the same Product allowed but remove the version installed in a specific path.

### DISCUSSION THREAD

**Discussion by OreCB** [wixtoolset/issues#8121] [Questions] [answered]
Hello,

I have a situation where we allow the user to have different versions of our Product installed in the same computer.
For example:
We have version 1.0 .0 installed in “C:\Product\Main\”
We have version 2.0.0 installed in “C:\Product\V2\”
These two versions can coexist.

If the user tries to install a new msi selecting a path that is already being used, we need to uninstall the specific version that is there, before we install the new one.
For example:
The user wants to install version 3.0.0 in the same path where version 1.0.0 is “C:\Product\Main\”
I need to uninstall version 1.0.0 as part of my installation of version 3.0.0.

I have tried a Custom Action where I find the Product Code of the version that is already in that path and try to uninstall it, but I get an error because there is an installation in progress, the one I just started.
I tried moving the execution of that Custom Action as early as possible in the installation process, but nothing worked.

Is there any other approach to this issue that you can suggest?
I cannot create a Bundle. The installation file must be a .msi

Thanks

**Comment by chrpai** [comment]
MSI has a mutex that prevents one msi from calling another msi in certain circumstances. This is one of them.  The best you could do author a major upgrade rule that doesn't find any then use the custom action to put the product code into the property the major upgrade is using.  This way RemoveExistingProducts should do the uninstall for you.

Have you just considered just blocking the install and saying  path already in use please pick another location?

**Reply by OreCB** [reply]
Hello, 
Thank you for your replay.
I will try what you suggested.
I tried blocking the install when using the path but got some push back from clients.
Thanks again.

**Comment by OreCB** [comment, marked as answer]
Hello,
An update on this issue.
I was finally able to get it to work. Using a C# DLL that read a file that contained the Product Code and assign it to the Property I used in the Upgrade. I ran the CA after the “FindRelatedProducts” and before the “RemoveExistingProducts”. Everything worked as expected. 
Thank you.

Note:
I tried to do it first by using the RegistrySearch or FileSearch to update the property, but the order where the property is populated was never good. If someone reading this has a similar issue. Don't waste days like I did. Use the CA with a C# DLL. It is super easy to implement. The HeatWave C# Custom Action Project does 75% of the work for you.
